core/src/main/java/hudson/model/Actionable.java

```diff
@@ -73,13 +74,23 @@
      */
     @Deprecated
 	public List<Action> getActions() {
-		if(actions == null) {
-			synchronized (this) {
-				if(actions == null) {
-					actions = new CopyOnWriteArrayList<Action>();
-				}
-			}
-		}
+        if(actions == null) {
+            synchronized (this) {
+                if (actions == null) {
+                    actions = new CopyOnWriteArrayList<Action>();
+                } else {
+                    Iterator actionIterator = actions.iterator();
```

Comment:
Use `Iterator<Action>` and then you can get rid of the cast below (as well as `rawtypes` warnings).